### PR TITLE
Add kubernetes_crud role

### PR DESCRIPTION
**Pull request by gabrielecerami**
this role a is a generic interface between ansible and batch deployments
of kuberneted manifests.
It allows to write steps in a declarative way avoiding duplication of
common parameters such as kubeconfig or context.
This role will be used by the upcoming rewrite of role to deploy prow
and its dependencies.

Signed-off-by: Gabriele Cerami <gcerami@redhat.com>

### CODE REVIEW

**Comment by gabrielecerami** [comment]
/assign @danielBelenky 

The job that tests the role passes. Launches kubevirtCI using docker in docker and tests the role with molecule.

File: images/kubevirt-ci-testenv/Dockerfile
Comment (dhiller) [inline images/kubevirt-ci-testenv/Dockerfile:4]: I guess you can remove this?

File: github/ci/prow/files/jobs/kubevirt/project-infra/project-infra-presubmits.yaml
Comment (dhiller) [inline github/ci/prow/files/jobs/kubevirt/project-infra/project-infra-presubmits.yaml:90]: Why is it necessary to start the docker daemon manually?

**Review by dhiller — COMMENTED** [review]
IIUC correctly this change introduces the framework with which we can test a deployment of prow on a kubevirtci cluster in prow. Looks good to me!

Just a question: I guess you will proceed by including the existing ansible deployment as further steps into the `converge` and add `validate` steps after this one is merged? How much will you need to rewrite for that step?

File: go.sum
Comment (gabrielecerami) [reply go.sum:545]: Yes, despite my attempt to not include it ...

File: images/kubevirt-ci-testenv/Dockerfile
Comment (gabrielecerami) [reply images/kubevirt-ci-testenv/Dockerfile:4]: I guess I can, if everything will be tested with debian upstream.

File: github/ci/prow/files/jobs/kubevirt/project-infra/project-infra-presubmits.yaml
Comment (gabrielecerami) [reply github/ci/prow/files/jobs/kubevirt/project-infra/project-infra-presubmits.yaml:90]: In some of my tests the docker daemon was not started with the container. But I changed a few things since. For example the docker file doesn't start anymore on an old version of bootstrap. I can add an additional test removing it, and see if it's really needed.

**Comment by gabrielecerami** [comment]
> IIUC correctly this change introduces the framework with which we can test a deployment of prow on a kubevirtci cluster in prow. Looks good to me!

> Just a question: I guess you will proceed by including the existing ansible deployment as further steps into the `converge` and add `validate` steps after this one is merged? How much will you need to rewrite for that step?

Not exactly. This will be a boilerplate for every other test that includes kubernetes-through-ansible activities.
The prow deployment tasks will be on another role, a separate role called prow-deploy. It will include this role here as the way to apply kubernetes manifest through ansible, but it will have its own set of tests, molecule directory, prepare, converge, and verify playbook.

File: github/ci/prow/files/jobs/kubevirt/project-infra/project-infra-presubmits.yaml
Comment (gabrielecerami) [reply github/ci/prow/files/jobs/kubevirt/project-infra/project-infra-presubmits.yaml:90]: Job succeded, removing

File: go.sum
Comment (gabrielecerami) [reply go.sum:545]: This is recreated every time I launch configcheck ... and I continue to accidentaly add it.

File: images/kubevirt-ci-testenv/Dockerfile
Comment (dhiller) [reply images/kubevirt-ci-testenv/Dockerfile]: No, I menat you can remove this commented out line.

File: github/ci/prow/files/jobs/kubevirt/project-infra/project-infra-presubmits.yaml
Comment (dhiller) [inline github/ci/prow/files/jobs/kubevirt/project-infra/project-infra-presubmits.yaml:83]: Maybe you can move this into the README?

**Review by dhiller — CHANGES_REQUESTED** [review]
All in all looks very good! One nit though. Decide yourself if you want to do this or remove the hold otherwise.

/lgtm
/approve
/hold

File: github/ci/prow/files/jobs/kubevirt/project-infra/project-infra-presubmits.yaml
Comment (gabrielecerami) [reply github/ci/prow/files/jobs/kubevirt/project-infra/project-infra-presubmits.yaml:83]: You're right, but everything looks so ready that I don't want to ruin this moment.
I'll move the comment into README with the next patch.

**Comment by gabrielecerami** [comment]
/hold cancel

Thanks for the review!

**Pull request merged**
